libportability-gfx/src/impls.rs

```diff
@@ -1772,14 +1788,113 @@ pub extern "C" fn gfxCreateGraphicsPipelines(
 }
 #[inline]
 pub extern "C" fn gfxCreateComputePipelines(
-    device: VkDevice,
-    pipelineCache: VkPipelineCache,
+    gpu: VkDevice,
+    _pipelineCache: VkPipelineCache,
     createInfoCount: u32,
     pCreateInfos: *const VkComputePipelineCreateInfo,
     _pAllocator: *const VkAllocationCallbacks,
     pPipelines: *mut VkPipeline,
 ) -> VkResult {
-    unimplemented!()
+    let infos = unsafe {
+        slice::from_raw_parts(pCreateInfos, createInfoCount as _)
+    };
+
+    let mut shader_stages = Vec::with_capacity(infos.len());
+
+    // Collect all information which we will borrow later. Need to work around
+    // the borrow checker here.
+    // TODO: try to refactor it once we have a more generic API
+    for info in infos {
+        let name = unsafe { CStr::from_ptr(info.stage.pName).to_owned() };
+        let specialization = unsafe { info.stage
+            .pSpecializationInfo
+            .as_ref()
+            .map(|specialization| {
+                let _data = slice::from_raw_parts(
+                    specialization.pData,
+                    specialization.dataSize as _,
+                );
+                let entries = slice::from_raw_parts(
+                    specialization.pMapEntries,
+                    specialization.mapEntryCount as _,
+                );
+
+                entries
+                    .into_iter()
+                    .map(|_entry| {
+                        // Currently blocked due to lack of specialization type knowledge
+                        unimplemented!()
+                    })
+                    .collect::<Vec<pso::Specialization>>()
+            })
+            .unwrap_or(vec![])
+        };
+
+        shader_stages.push((
+            name.into_string().unwrap(),
+            specialization,
+        ));
+    }
+
+    let descs = infos.into_iter().zip(&shader_stages).map(|(info, &(ref entry, ref specialization))| {
+        let shader = pso::EntryPoint {
+            entry,
+            module: &*info.stage.module,
+            specialization,
+        };
+        let layout = &*info.layout;
+
+        let flags = {
+            let mut flags = pso::PipelineCreationFlags::empty();
```

Comment:
should use `mem::transmute`.
I guess you reused the code from the graphics pipeline creation. Therefore we can also address it in a followup.